src/net-questdb-client/Ingress/Senders/HttpSender.cs

```diff
@@ -0,0 +1,394 @@
+
+        _handler.ConnectTimeout = Options.auth_timeout;
+        _handler.PreAuthenticate = true;
+
+        _client = new HttpClient(_handler);
+        var uri = new UriBuilder(Options.protocol.ToString(), Options.Host, Options.Port);
+        _client.BaseAddress = uri.Uri;
+        _client.Timeout = Timeout.InfiniteTimeSpan;
+
+        if (!string.IsNullOrEmpty(Options.username) && !string.IsNullOrEmpty(Options.password))
+        {
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
+                Convert.ToBase64String(Encoding.ASCII.GetBytes($"{Options.username}:{Options.password}")));
+        }
+        else if (!string.IsNullOrEmpty(Options.token))
+        {
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", Options.token);
+        }
+
+        return this;
+    }
+
+    /// <summary>
+    ///     Creates a new HTTP request with appropriate encoding and timeout.
+    /// </summary>
+    /// <returns></returns>
+    private (HttpRequestMessage, CancellationTokenSource?) GenerateRequest(CancellationToken ct = default)
+    {
+        var request = new HttpRequestMessage(HttpMethod.Post, "/write")
+            { Content = new BufferStreamContent(Buffer) };
+        request.Content.Headers.ContentType = new MediaTypeHeaderValue("text/plain") { CharSet = "utf-8" };
+        request.Content.Headers.ContentLength = Buffer.Length;
+        var cts = CancellationTokenSource.CreateLinkedTokenSource(ct);
+        cts.CancelAfter(CalculateRequestTimeout());
+        return (request, cts);
+    }
+
+    /// <summary>
+    ///     Calculate the request timeout.
+    /// </summary>
+    /// <remarks>
+    ///     Large requests may need more time to transfer the data.
+    ///     This calculation uses a base timeout (<see cref="QuestDBOptions.request_timeout" />), and adds on
+    ///     extra time corresponding to the expected transfer rate (<see cref="QuestDBOptions.request_min_throughput" />)
+    /// </remarks>
+    /// <returns></returns>
+    private TimeSpan CalculateRequestTimeout()
+    {
+        return Options.request_timeout
+               + TimeSpan.FromSeconds(Buffer.Length / (double)Options.request_min_throughput);
+    }
+
+    /// <inheritdoc cref="ISender.Transaction(ReadOnlySpan&lt;char&gt;)"/>
+    public ISender Transaction(ReadOnlySpan<char> tableName)
+    {
+        if (WithinTransaction)
+        {
+            throw new IngressError(ErrorCode.InvalidApiCall,
+                "Cannot start another transaction - only one allowed at a time.");
+        }
+        
+        if (Length > 0)
+        {
+            throw new IngressError(ErrorCode.InvalidApiCall,
+                "Buffer must be clear before you can start a transaction.");
+        }
+
+        Buffer.Transaction(tableName);
+        return this;
+    }
+
+    /// <inheritdoc cref="ISender.Commit()"/>
+    public void Commit()
+    {
+        CommitAsync().Wait();
```

Comment:
This is a bad smell. There is a reason HttpClient has both `Send()` and `SendAsync()`, in certain execution contexts it can lead to deadlock.

Please implement Send() using the Http client `Send()` Method without using async/await.